maitake/src/wait/dataqueue/tests.rs

```diff
@@ -0,0 +1,272 @@
+use super::*;
+
+#[cfg(all(not(loom), feature = "alloc"))]
+mod alloc {
+    use super::*;
+    use crate::loom::sync::Arc;
+    use crate::scheduler::Scheduler;
+    use core::sync::atomic::{AtomicUsize, Ordering};
+
+    //     #[test]
+    //     fn close() {
+    //         crate::util::trace_init();
+    //         static COMPLETED: AtomicUsize = AtomicUsize::new(0);
+
+    //         let scheduler = Scheduler::new();
+    //         let q = Arc::new(DataWaitQueue::new());
+
+    //         const TASKS: usize = 10;
+
+    //         for _ in 0..TASKS {
+    //             let wait = q.wait_owned();
+    //             scheduler.spawn(async move {
+    //                 wait.await.expect_err("dropping the queue must close it");
+    //                 COMPLETED.fetch_add(1, Ordering::SeqCst);
+    //             });
+    //         }
+
+    //         let tick = scheduler.tick();
+
+    //         assert_eq!(tick.completed, 0);
+    //         assert_eq!(COMPLETED.load(Ordering::SeqCst), 0);
+    //         assert!(!tick.has_remaining);
+
+    //         q.close();
+
+    //         let tick = scheduler.tick();
+
+    //         assert_eq!(tick.completed, TASKS);
+    //         assert_eq!(COMPLETED.load(Ordering::SeqCst), TASKS);
+    //         assert!(!tick.has_remaining);
+    //     }
+
+    #[test]
+    fn wake_one() {
```

Comment:
@hawkw I did re-enable this test, which also shows sort of how you'd use this in the "normal" case. This might be worth taking a peek at first.